src/scripts/components/widgets/modals/ModalStake.js

```diff
@@ -35,41 +40,48 @@ const Footer = styled.div`
   width: 100%;
 `
 
-type Props = {
-  videoId: String,
-  user: UserRecord,
-  onSuccess: () => void
-}
-
 class ModalStake extends Component<Props, Object> {
   onSubmit: (e: Object) => void
   constructor (props: Props) {
     super(props)
+    const selectedVideo = this.props.selectedVideo
     this.state = {
+      id: selectedVideo.id,
+      title: selectedVideo.title,
+      description: selectedVideo.description,
+      // FIXME: we are not editing duration, so we do not need to store it in the state
+      duration: selectedVideo.duration,
+      author: selectedVideo.author,
       errorMessage: false,
-      agreedTOC: false // TODO
+      agreedTOC: false // TODO,
     }
     this.onSubmit = this.onSubmit.bind(this)
-    // this.handlePublish = this.handlePublish.bind(this)
-    // this.handlePublish = this.handlePublish.bind(this)
   }
 
   onSubmit (event: Object) {
     event.preventDefault()
 
     paratii.eth.tcr
       .checkEligiblityAndApply(
-        this.props.videoId.toString(),
+        this.props.videoId.id.toString(),
```

Comment:
Why is the prop called `videoId` if it has a property called `id`? Shouldn't the prop be called `video`?

Reply:
Done https://github.com/Paratii-Video/paratii-portal/pull/301/commits/49f681e2a5dc7801ec9e0e4aa422af395946a54b